optuna/pruners/threshold.py

```diff
@@ -0,0 +1,59 @@
+from optuna.pruners import BasePruner
+
+from optuna.type_checking import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import Optional
+
+    from optuna.structs import FrozenTrial
+    from optuna.study import Study
+
+
+class ThresholdPruner(BasePruner):
+    """Pruner to detect abnormal metrics of the trials.
```

Comment:
What do you think of using a different term than "abnormal". One might for instance want to use this pruner to prune trials that have reached a sufficiently high accuracy, in which case "abnormal" might sound misleading. A suggestion could be
```suggestion
    """Pruner to detect outlying metrics of the trials.
```

Reply:
Thank you for the suggestion. :memo: